packages/react/src/forms/RadioGroup/RadioGroup.test.tsx

```diff
@@ -0,0 +1,153 @@
+import React, {render, cleanup} from '@testing-library/react'
+import '@testing-library/jest-dom'
+import {axe, toHaveNoViolations} from 'jest-axe'
+
+import {RadioGroup} from './RadioGroup'
+import {Radio} from '../Radio'
+import {FormControl} from '../FormControl'
+
+expect.extend(toHaveNoViolations)
+
+describe('RadioGroup', () => {
+  afterEach(cleanup)
+
+  it('renders a radio group correctly into the document', () => {
+    const {getByLabelText, getByRole, getByText} = render(
+      <RadioGroup>
+        <RadioGroup.Label>Choices</RadioGroup.Label>
+        <RadioGroup.Caption>You can only pick one</RadioGroup.Caption>
+        <FormControl>
+          <FormControl.Label>Choice one</FormControl.Label>
+          <Radio name="demo" value="one" defaultChecked />
+        </FormControl>
+        <FormControl>
+          <FormControl.Label>Choice two</FormControl.Label>
+          <Radio name="demo" value="two" />
+        </FormControl>
+        <FormControl>
+          <FormControl.Label>Choice three</FormControl.Label>
+          <Radio name="demo" value="three" />
+        </FormControl>
+
+        <RadioGroup.Validation variant="success">Great job!</RadioGroup.Validation>
+      </RadioGroup>,
+    )
+
+    expect(getByRole('group', {name: 'Choices'})).toBeInTheDocument()
+    expect(getByText('You can only pick one')).toBeInTheDocument()
+    expect(getByLabelText('Choice one')).toBeInTheDocument()
+    expect(getByLabelText('Choice two')).toBeInTheDocument()
+    expect(getByLabelText('Choice three')).toBeInTheDocument()
+    expect(getByText('Great job!')).toBeInTheDocument()
+  })
+
+  it('has no a11y violations', async () => {
+    const {container} = render(
+      <RadioGroup>
+        <RadioGroup.Label>Choices</RadioGroup.Label>
+        <RadioGroup.Caption>You can only pick one</RadioGroup.Caption>
+        <FormControl>
+          <FormControl.Label>Choice one</FormControl.Label>
+          <Radio name="demo" value="one" defaultChecked />
+        </FormControl>
+        <FormControl>
+          <FormControl.Label>Choice two</FormControl.Label>
+          <Radio name="demo" value="two" />
+        </FormControl>
+        <FormControl>
+          <FormControl.Label>Choice three</FormControl.Label>
+          <Radio name="demo" value="three" />
+        </FormControl>
+
+        <RadioGroup.Validation variant="success">Great job!</RadioGroup.Validation>
+      </RadioGroup>,
+    )
+    const results = await axe(container)
+
+    expect(results).toHaveNoViolations()
+  })
+
+  describe('aria-describedby', () => {
```

Comment:
Nit and more of a style preference: do we need the nested describe block? Could we instead rename the title of each test case to be more descriptive?

Reply:
Not required at all, but I quite like it as it

- Keeps test names more focussed
- Logically groups related tests in the code
- Groups tests in the output

![image](https://github.com/user-attachments/assets/f052de71-c070-4b3d-b15b-3f28a60a3a3a)

Happy to remove it if you don't like it though

Reply:
Cool, updated 👍